encryptedlogging/src/test/kotlin/com/automattic/encryptedlogging/persistence/dao/EncryptedLogDaoTest.kt

```diff
@@ -0,0 +1,236 @@
+        database = Room.inMemoryDatabaseBuilder(context, EncryptedLogDatabase::class.java)
+            .build()
+        sut = database.encryptedLogDao
+    }
+
+    @After
+    @Throws(IOException::class)
+    fun closeDb() {
+        database.close()
+    }
+
+    @Test
+    fun `test insert encrypted log`() = runTest {
+        // Assert that there are no encrypted logs with the test uuid
+        assertThat(sut.getEncryptedLog(TEST_UUID)).isNull()
+
+        // Insert an encrypted log with uuid
+        val logToBeInserted = createTestEncryptedLogModel()
+        sut.insertEncryptedLog(logToBeInserted)
+
+        // Assert that the encrypted log from the DB is the same as the one we inserted (ignoring id)
+        assertThat(sut.getEncryptedLog(TEST_UUID))
+            .usingRecursiveComparison()
+            .ignoringFields("id")
+            .isEqualTo(logToBeInserted)
+    }
+
+    @Test
+    fun `test insert multiple encrypted logs`() = runTest {
+        // Assert that there are no encrypted logs with the test uuid
+        assertThat(sut.getEncryptedLog(TEST_UUID)).isNull()
+
+        // Insert an encrypted log with uuid
+        val uuidList = (1..5).map { "uuid-prefix-$it" }
+        val logsToBeInserted = uuidList.map { createTestEncryptedLogModel(uuid = it) }
+        sut.upsertEncryptedLogs(logsToBeInserted)
+
+        // Assert that the encrypted logs from the DB is the same as the ones we inserted (ignoring id)
+        uuidList.forEachIndexed { index, uuid ->
+            val log = sut.getEncryptedLog(uuid)
+            assertThat(log)
+                .usingRecursiveComparison()
+                .ignoringFields("id")
+                .isEqualTo(logsToBeInserted[index])
+        }
+    }
+
+    @Test
+    fun `test update encrypted log`() = runTest {
+        // Insert an initial encrypted log
+        val initialLog = createTestEncryptedLogModel()
+        sut.insertEncryptedLog(initialLog)
+        assertThat(sut.getEncryptedLog(TEST_UUID))
+            .usingRecursiveComparison()
+            .ignoringFields("id")
+            .isEqualTo(initialLog)
+
+        // Get the encrypted log from the database (to get the correct id)
+        val initialLogFromDb = sut.getEncryptedLog(TEST_UUID)
+        assertThat(initialLogFromDb).isNotNull
+
+        // Create a copy of the encrypted log by changing its upload state (which will be the common usage)
+        val newUploadState = EncryptedLogUploadState.UPLOADING
+        val updatedLog = initialLogFromDb?.copy(uploadStateDbValue = newUploadState.value)
+        updatedLog?.let { sut.upsertEncryptedLog(updatedLog) }
+
+        // Assert that the encrypted log in the DB is the one with the correct upload state
+        val updatedLogFromDB = sut.getEncryptedLog(TEST_UUID)
+        assertThat(requireNotNull(updatedLogFromDB?.uploadState)).isEqualTo(newUploadState)
+        // This verifies the expected state as well but separating the initial assertion is valuable to show intent
```

Comment:
I understand the intention, but IMO it's not needed. In the second assertion, the different `uploadState` will be highlighted. My suggestion would be to keep tests as simple and focused as possible

```suggestion

```

Reply:
Yea, I agree @wzieba, I just converted this test with minimal changes to keep it intact.

PS: [Same](https://github.com/Automattic/EncryptedLogging/pull/20#discussion_r2163837693)

Reply:
FYI: This is now done: 09d587eec492d6c8ad5a57a69f2e8d90e56cfadb